api/apps/api/src/modules/scenario-specification/__tests__/mark-as-calculated/scenario-specification-mark-as-calculated.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { getFixtures } from './fixtures';
+
+let fixtures: ReturnType<typeof getFixtures>;
+
+beforeEach(() => {
+  fixtures = getFixtures();
+});
+
+describe(`When marked one of pending calculations as done`, () => {
+  beforeEach(() => {
+    fixtures.scenarioSpec.markFeaturesAsCalculated([
+      fixtures.nonCalculatedFeatureOne,
+    ]);
+  });
+
+  test(`Then specification isn't yet activated`, () => {
+    fixtures.ThenSpecificationIsNotActivated();
+  });
+
+  describe(`When marking remaining pending feature`, () => {
+    beforeEach(() => {
+      fixtures.scenarioSpec.markFeaturesAsCalculated([
+        fixtures.nonCalculatedFeatureTwo,
+      ]);
+    });
+
+    test(`ThenSpecificationIsActivated`, () => {
+      fixtures.ThenSpecificationIsActivated();
+    });
+  });
```

Comment:
I'd keep it unnested to not reuse the previous state – it's harder to infer what is really given